File: docs/snippets/all/howto/ingestion-guide/log_rows.cpp
// Use the log APIs to log scalars over time.

#include <rerun.hpp>

#include <cmath>

const size_t NUM_STEPS = 100000;
const double COEFF = 10.0 / static_cast<double>(NUM_STEPS);

int main() {
    const auto rec = rerun::RecordingStream("rerun_example_log_rows");
    rec.to_stdout().exit_on_failure();

    // Log the data on a timeline called "step".
    for (int step = 0; step < NUM_STEPS; ++step) {
        // Set the `step` timeline in the logging context to the current time.
        rec.set_time_sequence("step", step);

        // Log a new row containing a single scalar.
        // This will inherit from the logging context, and thus be logged at the current `step`.
        rec.log("scalar", rerun::Scalar(std::sin(static_cast<double>(step) * COEFF)));
    }
}

File: docs/snippets/all/howto/ingestion-guide/log_rows.rs
//! Use the log APIs to log scalars over time.

const NUM_STEPS: usize = 100_000;
const COEFF: f64 = 10.0 / NUM_STEPS as f64;

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let rec = rerun::RecordingStreamBuilder::new("rerun_example_log_rows").stdout()?;

    for step in 0..NUM_STEPS {
        // Set the `step` timeline in the logging context to the current time.
        rec.set_time_sequence("step", step as i64);

        // Log a new row containing a single scalar.
        // This will inherit from the logging context, and thus be logged at the current `step`.
        rec.log("scalars", &rerun::Scalar::new((step as f64 * COEFF).sin()))?;
    }

    Ok(())
}
